### PR TITLE
add Okta SSO integration

### PR DESCRIPTION
**Pull request by hugovalente-pm**
##### Summary

with the imminent release of Okta SSO we need to have our integration docs setup, it is a prerequisite to get the app on OIN

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by car12o — APPROVED** [review]

File: integrations/cloud-authentication/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+      ### Netdata Configuration Steps
+      1. Click on the Space settings cog (located above your profile icon)
+      2. Click on the **Authentication** tab
+      3. On the Okta SSO card, click on **Configure**
+      4. Fill in the required credentials, you get them from **Okta Admin Portal**:
+      - **Issuer URL** you can get it from your profile icon on top, e.g. `https://company-name.okta.com`
+      - **Client ID** you can get it from **General** tab on application you configured on Okta
+      - **Client Secret** you can get it from **General** tab on application you configured on Okta
+      Note: [Okta - Find your application credentials](https://developer.okta.com/docs/guides/find-your-app-credentials/main/)
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yml:45]: This requires an indentation of 4 spaces.

**Comment by Ancairon** [comment]
@hugovalente-pm don't we need code for this too? I see you have added a new file, but that won't do it I think..

We need to alter the script if we want to create a new category, is this the need here?

**Comment by Ancairon** [comment]
I am removing the approval in this PR, because at the moment if we merge this nothing will happen and probably something will break. 

I am looking at the code to see how we can add this to the workflow.

@hugovalente-pm what schema did you use to make the file? I will also rename it to yaml instead of yml

**Comment by hugovalente-pm** [comment]
> @hugovalente-pm what schema did you use to make the file? I will also rename it to yaml instead of yml

used the same as this https://github.com/netdata/netdata/blob/master/integrations/cloud-notifications/metadata.yaml since the output of the integration card we want is the same

**Comment by Ancairon** [comment]
we want another schema tough, because filling in 

```
    notification_description: "Integrate your organizations Okta account with Netdata to better manage your team's
    access controls to Netdata Cloud."
```

is looking wrong.. Anyways, we need coding too, so I will make a new schema too. I take the PR from here and will req review once I am done

**Comment by hugovalente-pm** [comment]
> is looking wrong.. Anyways, we need coding too, so I will make a new schema too. I take the PR from here and will req review once I am done

thanks @Ancairon 🙌 sorry or the work, was trying to see if I could take this one myself

**Comment by Ancairon** [comment]
no worries, sadly the process is not automatic and does need some code, so it would require someone who engineered this anyways, I hope I can get this done today

File: integrations/cloud-authentication/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - An Okta account
+      - A Netdata Cloud account
+      - Access to the space as an administrator
+      - Space needs to be on Business plan or higher
+
+      ### Setting up Okta 
+      Steps needed to be done on Okta Admin Portal:
+      1. Click on **Applications** tab and choose to **Browse App Catalogue**
+      2. Find Netdata's preconfigured app for easy setup and click **Add Integration**
+      3. Give the app, that will be in your apps dashboard, the prefered **Application label**
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yaml:30]:
```suggestion
      3. Give the app, that will be in your apps dashboard, the preferred **Application label**
```

File: integrations/cloud-authentication/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      and click **Next** to move to the Sign-On options tab
+      4. In the **Sign-On Options** all the values we expect are already filled and no additional data is required
+      5. Click **Done**. You are be able to go back and edit any fields later if need be
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yaml:33]:
```suggestion
      5. Click **Done**. You are able to go back and edit any fields later if need be
```

File: integrations/cloud-authentication/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      6- Go to the **Assignments** tab and enter the People or Group assignments as per your
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yaml:34]:
```suggestion
      6. Go to the **Assignments** tab and enter the People or Group assignments as per your
```

**Comment by Ancairon** [comment]
@hugovalente-pm

1. > Give the app, that will be in your apps dashboard, the prefered **Application label**
      and click **Next** to move to the Sign-On options tab

    what does "give the app mean"?

2. you will also need to add the okta logo with the filename you specified I think in [here](https://github.com/netdata/website/tree/master/themes/tailwind/static/img)

3. and I think you should also coordinate with John Kapatzakis in order to see if the app is ready to accept such an integration (I don't think that is automatic either)

4. Also let me know where you want this on Learn, as I would need to include (or not) the page in Learn

do any edits on the **YAML** file and not on the .md, that will be auto-generated afterwards, we don't touch it now

File: integrations/cloud-authentication/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  overview:
+    authentication_description: "Integrate your organization's Okta account with Netdata to better manage your team's access controls to Netdata Cloud."
+    authentication_limitations: ''
+  setup:
+    description: |
+      ### Prerequisites
+      - An Okta account
+      - A Netdata Cloud account
+      - Access to the Space as an administrator
+      - Space needs to be on Business plan or higher
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yaml:23]:
```suggestion
      - Space needs to be on the Business plan or higher
```

File: integrations/cloud-authentication/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+      ### Setting up Okta
+      Steps needed to be done on Okta Admin Portal:
+      1. Click on **Applications** tab and choose to **Browse App Catalogue**
+      2. Find Netdata's preconfigured app for easy setup and click **Add Integration**
+      3. Give the app, that will be in your apps dashboard, the preferred **Application label** and click **Next** to move to the Sign-On options tab
+      4. In the **Sign-On Options** all the values we expect are already filled and no additional data is required
+      5. Click **Done**. You are able able to go back and edit any fields later if need be
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yaml:31]:
```suggestion
      5. Click **Done**. You are able to go back and edit any fields later if need be
```

File: integrations/cloud-authentication/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+      ### Netdata Configuration Steps
+      1. Click on the Space settings cog (located above your profile icon)
+      2. Click on the **Authentication** tab
+      3. On the Okta SSO card, click on **Configure**
+      4. Fill in the required credentials, you get them from **Okta Admin Portal**:
+          - **Issuer URL** you can get it from your profile icon on top, e.g. `https://company-name.okta.com`
+          - **Client ID** you can get it from **General** tab on application you configured on Okta
+          - **Client Secret** you can get it from **General** tab on application you configured on Okta
+          Note: [Okta - Find your application credentials](https://developer.okta.com/docs/guides/find-your-app-credentials/main/)
```
Comment (ilyam8) [inline integrations/cloud-authentication/metadata.yaml:42]:
```suggestion
      4. Fill in the [required credentials](https://developer.okta.com/docs/guides/find-your-app-credentials/main/), you get them from **Okta Admin Portal**:
          - **Issuer URL** you can get it from your profile icon on top, e.g. `https://company-name.okta.com`
          - **Client ID** you can get it from **General** tab on application you configured on Okta
          - **Client Secret** you can get it from **General** tab on application you configured on Okta
```

**Comment by hugovalente-pm** [comment]
@Ancairon 

>> 1. Give the app, that will be in your apps dashboard, the prefered Application label
>> and click Next to move to the Sign-On options tab
> what does "give the app mean"?

in Okta Admin panel you create an "app" for the integration you want to setup and you need to give it a name (there is a default one that you can edit)

https://github.com/netdata/netdata/assets/82235632/c0ee9ea9-fc6d-409a-9b84-4d6406ac8fbe

once you set it up it will be available on your Apps dashboard and have that display name
image

> 2. you will also need to add the okta logo with the filename you specified I think in [here](https://github.com/netdata/website/tree/master/themes/tailwind/static/img)

was going to do it as soon as I saw this ready

> 3. and I think you should also coordinate with John Kapatzakis in order to see if the app is ready to accept such an integration (I don't think that is automatic either)

for submitting this to Okta we need it either on netdata.cloud or learn, app can come later even better once we have the OIN ready

> 4. Also let me know where you want this on Learn, as I would need to include (or not) the page in Learn

should be under https://learn.netdata.cloud/docs/netdata-cloud/ since it is related with Netdata Cloud really and should be near by the "Role-based access model" probably before that, if possible

**Comment by Ancairon** [comment]
Then @hugovalente-pm do you want to say "give the app a name"?

**Comment by hugovalente-pm** [comment]
> Then @hugovalente-pm do you want to say "give the app a name"?

we should kee the "Application label" there since it is how Okta names this attribute, probably it could be reworded
> Give the app the preferred Application label, for when it is displayed in your apps dashboard, 
and click Next to move to the Sign-On options tab

> For the website I am not sure who was doing the integration stuff before

it was John, I can ping him once this is ready

**Comment by Ancairon** [comment]
@hugovalente-pm I added the edit -now I get the sentence- , I think it is more clear now

**Review by Ancairon — APPROVED** [review]

**Pull request merged**